### PR TITLE
DOC: add agg nunique example

### LINKED ISSUE #6280: doc: add nunique example to custom aggregation
I was looking up how to do `nunique` on a `dask.dataframe` and https://stackoverflow.com/a/47931009/6046019 seems to be a solution.

I would like to add this to the bottom of https://docs.dask.org/en/latest/dataframe-groupby.html#aggregate as another example of a custom aggregation.

If there's a better way I will be happy to write that.

```
import pandas as pd
import pandas._testing as tm
import dask.dataframe as dd

d = {'col1': [1, 2, 3, 4], 'col2': [5, 6, 7, 8]}
df = pd.DataFrame(data=d)
expected = df.groupby(['col1']).agg({'col2':'nunique'})

d_df = dd.from_pandas(df, npartitions=2)

nunique = dd.Aggregation(
    name="nunique",
    chunk=lambda s : s.apply(lambda x: list(set(x))),
    agg=lambda s0 : s0._selected_obj.groupby(level=list(range(s0._selected_obj.index.nlevels))).sum(),
    finalize=lambda s1 : s1.apply(lambda final: len(set(final))),
    )

actual = d_df.groupby(['col1']).agg({'col2':nunique}).compute()

tm.assert_equal(actual, expected)
```

### PR DESCRIPTION
**Pull request by raybellwaves**
- [NA] Tests added / passed
- [NA] Passes `black dask` / `flake8 dask`

Closes #6280 

image

### CODE REVIEW

File: docs/source/dataframe-groupby.rst [hunk 1]
```diff
@@ -220,3 +220,18 @@ Pandas' ``groupby('g').agg(list)``:
...
+
+    >>> nunique = dd.Aggregation(
+    ...     name="nunique",
+    ...     chunk=lambda s : s.apply(lambda x: list(set(x))),
+    ...     agg=lambda s0 : s0._selected_obj.groupby(
+    ...         level=list(range(s0._selected_obj.index.nlevels))
+    ...     ).sum(),
+    ...     finalize=lambda s1 : s1.apply(lambda final: len(set(final))),
+    )
+    >>> df.groupby('g').agg({'a':nunique, 'b':nunique})
```
Comment (jcrist) [inline docs/source/dataframe-groupby.rst:237]: This should probably be `ddf` instead. And `'g'` isn't an existing column in the example dataframe - could add one (I'd probably call it `'c'` instead, or use `'a'` again perhaps? Not sure what makes sense.

I noticed that the previous example has the same issue, if you could fix that one as well that'd be useful. Ideally our docs examples all run copy-pastable.

File: docs/source/dataframe-groupby.rst [hunk 2]
```diff
@@ -220,3 +220,18 @@ Pandas' ``groupby('g').agg(list)``:
...
+column you can use:
+
+.. code-block:: python
+
+    >>> nunique = dd.Aggregation(
+    ...     name="nunique",
+    ...     chunk=lambda s : s.apply(lambda x: list(set(x))),
+    ...     agg=lambda s0 : s0._selected_obj.groupby(
+    ...         level=list(range(s0._selected_obj.index.nlevels))
+    ...     ).sum(),
```
Comment (jcrist) [inline docs/source/dataframe-groupby.rst:234]: Python doesn't support multi-line lambdas, so this is a syntax error. Perhaps define a new function, or drop the line-wrap?

Comment (jcrist) [reply docs/source/dataframe-groupby.rst:234]: It also feels a bit odd to use a hidden attribute (`_selected_obj`) here. cc @TomAugspurger for thoughts.

Comment (quasiben) [reply docs/source/dataframe-groupby.rst:236]: I think we should be able to do this with `obj` instead of `_selected_obj`

**Comment by raybellwaves** [comment]
Friendly ping

File: dask/dataframe/groupby.py [hunk 3]
```diff
@@ -1792,6 +1792,17 @@ def __init__(self, df, by=None, slice=None, **kwargs):
 
     @derived_from(pd.core.groupby.SeriesGroupBy)
     def nunique(self, split_every=None, split_out=1):
+        """
+        Examples
+        --------
+
```
Comment (jcrist) [inline dask/dataframe/groupby.py:1798]: Looks like you've failed the linting checks due to trailing whitespace here: https://travis-ci.org/github/dask/dask/jobs/707841008

```suggestion

```

**Comment by jsignell** [comment]
This looks great! Thanks @raybellwaves!

**Pull request merged**